### DISCUSSION TITLE
Why trivy has much less vulnerabilities discoveryed, compare to dependency-check ?

### DISCUSSION THREAD

**Discussion by WestFarmer** [aquasecurity/trivy#7572] [Q&A] [answered]
### Question

I have a simple spring-boot project, test it against trivy, grype and dependency-check

here is results:
trivy found 10 vulnerabilities
grype found 14
dependency-check found 45

trivy :
```
CVE-2023-6841
CVE-2024-1722
CVE-2024-22262
CVE-2024-25710
CVE-2024-26308
CVE-2024-34750
CVE-2024-38816
CVE-2024-40094
CVE-2024-7260
CVE-2024-7318
GHSA-9vm7-v8wj-3fqw
```
grype:
```
CVE-2022-31197
CVE-2023-6841
CVE-2024-1597
CVE-2024-1722
CVE-2024-21742
CVE-2024-22262
CVE-2024-25710
CVE-2024-26308
CVE-2024-34750
CVE-2024-38816
CVE-2024-40094
CVE-2024-7260
CVE-2024-7318
GHSA-9vm7-v8wj-3fqw
```

dependency-check:
```
CVE-2019-20851 CVE-2020-13891 CVE-2020-26945
CVE-2021-37860 CVE-2021-37861 CVE-2021-37864
CVE-2021-37865 CVE-2022-0708 CVE-2022-1002
CVE-2022-1003 CVE-2022-2406 CVE-2022-2408
CVE-2022-4044 CVE-2023-1562 CVE-2023-2514
CVE-2023-25330 CVE-2023-27263 CVE-2023-27264
CVE-2023-35075 CVE-2023-35116 CVE-2023-3615
CVE-2023-40703 CVE-2023-43754 CVE-2023-45223
CVE-2023-47168 CVE-2023-47865 CVE-2023-48268
CVE-2023-48369 CVE-2023-5522 CVE-2023-5967
CVE-2023-5968 CVE-2023-5969 CVE-2023-6134
CVE-2023-6202 CVE-2023-6291 CVE-2023-6927
CVE-2023-7114 CVE-2024-21742 CVE-2024-22262
CVE-2024-25710 CVE-2024-26308 CVE-2024-38809
CVE-2024-38816 CVE-2024-40094 CVE-2024-4629
```

it seems dc found many old cves, while trivy found most recent cves.

any comments on this ?

 

### Target

Git Repository

### Scanner

Vulnerability

### Output Format

JSON

### Mode

Standalone

**Comment by DmitriyLewen** [comment]
Hello @WestFarmer 

I can't answer this question for sure.

Perhaps your repository contains file types that are not supported by Trivy (see https://aquasecurity.github.io/trivy/v0.55/docs/coverage/language/#supported-languages)
Or the advisory databases we use don't contain these vulnerabilities (see https://aquasecurity.github.io/trivy/v0.55/docs/scanner/vulnerability/#data-sources_1)

Regards, Dmitriy

**Comment by WestFarmer** [comment, marked as answer]
it was caused by dependency-check false positive, 31 vuls related to mattermost rest client, but those vuls applied to mattermost mobile App..

thanks.



---- Replied Message ----
| From | ***@***.***> |
| Date | 09/24/2024 19:07 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [aquasecurity/trivy] Why trivy has much less vulnerabilities discoveryed, compare to dependency-check ? (Discussion #7572) |

Hello @WestFarmer

I can't answer this question for sure.

Perhaps your repository contains file types that are not supported by Trivy (see https://aquasecurity.github.io/trivy/v0.55/docs/coverage/language/#supported-languages)
Or the advisory databases we use don't contain these vulnerabilities (see https://aquasecurity.github.io/trivy/v0.55/docs/scanner/vulnerability/#data-sources_1)

Regards, Dmitriy

**Reply by knqyf263** [reply]
When I saw dependency-check before, it was using fuzzy matching (I may be wrong). While it may be predictable without completely identifying the package name, it may detect the wrong package name. Because different tools have different policies, it is difficult to compare only the number of vulnerabilities detected and the number of false positives and false negatives must be carefully compared.
